src/utils/DependsOnManager.ts

```diff
@@ -1,82 +1,111 @@
-  toggleDependentFacet: (dependentFacet: Component) => void;
-  element: HTMLElement;
-  root: HTMLElement;
-  dependsOn: string;
-  id: string;
-  queryStateModel: QueryStateModel;
-  bind: ComponentEvents;
+  ref: Component;
+}
+
+export interface IDependentFacetCondition {
+  (facet: Component): boolean;
 }
 
 export class DependsOnManager {
+  private parentFacet: Component;
+
   constructor(private facet: IDependentFacet) {
-    this.facet.bind.onRootElement(QueryEvents.newQuery, () => this.handleNewQuery());
-    this.updateVisibilityBasedOnDependsOn();
+    this.facet.ref.bind.onRootElement(QueryEvents.newQuery, () => this.handleNewQuery());
+
+    if (this.getDependsOn(this.facet.ref)) {
+      $$(this.facet.ref.element).addClass('coveo-hidden');
+      this.setupDependentFacet();
+    }
   }
 
-  public listenToParentIfDependentFacet() {
-    if (!this.isDependentFacet) {
-      return;
+  private setupDependentFacet() {
+    this.parentFacet = this.getParentFacet(this.facet.ref);
+
+    if (this.parentFacet) {
+      this.facet.ref.bind.onQueryState(MODEL_EVENTS.CHANGE, undefined, () => this.resetIfConditionUnfullfiled());
     }
-    this.facet.bind.onQueryState(MODEL_EVENTS.CHANGE, undefined, () => this.resetIfParentFacetHasNoSelectedValues());
   }
 
-  public updateVisibilityBasedOnDependsOn() {
-    if (!this.isDependentFacet) {
-      return;
+  private resetIfConditionUnfullfiled() {
+    const condition = this.getDependsOnCondition(this.facet.ref);
+
+    if (!condition(this.parentFacet)) {
+      this.facet.reset();
     }
+  }
 
-    $$(this.facet.element).toggleClass('coveo-hidden', !this.parentFacetHasSelectedValues);
+  private getId(component: Component) {
+    const id = component.options.id;
+    return id ? `${id}` : null;
   }
 
-  private get dependentFacets() {
-    const allFacets = ComponentsTypes.getAllFacetsInstance(this.facet.root);
-    return allFacets.filter(facet => this.facet.id === facet.options.dependsOn);
+  private getDependsOn(component: Component) {
+    const dependsOn = component.options.dependsOn;
+    return dependsOn ? `${dependsOn}` : null;
   }
 
-  public get hasDependentFacets() {
-    return !!this.dependentFacets.length;
+  private getDependsOnCondition(component: Component): IDependentFacetCondition {
+    const conditionOption = component.options.dependsOnCondition;
+    return conditionOption && isFunction(conditionOption) ? conditionOption : () => this.parentHasSelectedValues(component);
   }
 
-  public get dependentFacetsHaveSelectedValues() {
-    return this.dependentFacets.some(facet => this.valuesExistForFacetWithId(facet.options.id));
+  private parentHasSelectedValues(component: Component) {
+    const parent = this.getParentFacet(component);
+    return parent && this.valuesExistForFacetWithId(this.getId(parent));
   }
 
-  private get isDependentFacet() {
-    return Utils.isNonEmptyString(this.facetDependsOnField);
+  private valuesExistForFacetWithId(facetId: string) {
```

Comment:
I'm thinking the name should probably be "facetIsActive", or "facetHasSelectedOrExcludedValues" ?

At first, I thought this was checking that the facet "has values", ie: is not empty.

Also, I think this could be a good public (and documented) methods to have on the different types of dynamic facets, instead of resolving that "indirectly" through query state model. I could easily see an integrator needing to get this information. And this is logic that should closely reside with the component.

eg:
```
class DynamicFacet {

  public hasActiveValues(): boolean
  public hasSelectedValues(): boolean // yeah, I know, we still don't allow to exclude values, but... you get the point
  public get selectedValues(): []IDynamicFacetValues
  public get excludedValues(): []IDynamicFacetValues
   // etc..
}
```
And again, those could be forced at the interface level (the one I described above in this PR)

Reply:
The default condition is checking if the parent facet has *selected* values, then it would display the child. That was how the original feature was coded. Showing after excluding a facet value doesn't make much sense. Besides, if a client wants that, they could use a custom condition.

QSM is great for working with that feature as it works with any type of filters. (Facet, DynamicFacet, SimpleFilter, etc.)

Going to add some method is every facet to tell the same thing as the QSM tells us would bloat the software a bit.